js/packages/ufg-client/test/unit/fetch-resources.spec.ts

```diff
@@ -72,6 +73,36 @@ describe('fetch-resource', () => {
     assert.deepStrictEqual(resource2, mockResource)
   })
 
+  it('fetch with concurrency limitation', async () => {
+    const mockResources = []
+    let count = 0
+    for (let i = 0; i < 10; i++) {
+      mockResources.push(
+        makeResource({
+          url: `http://something${i}`,
+        }),
+      )
+      nock(`http://something${i}`).get('/').once().reply(limitServerParallelRequests)
+    }
+
+    const fetchResource = makeFetchResource({fetchConcurrency: 5})
+    const resResources = await Promise.all(mockResources.map(resource => fetchResource({resource})))
+
+    assert.strictEqual(
+      resResources.filter(resource => {
+        utils.types.has(resource, `errorStatusCode`)
+      }).length,
+      0,
+    )
```

Comment:
Using filter + length is not the best way to get a number of elements that match your condition, but in this case, this is also not what you need. You actually want to know if there are any failed resources in the array. So I recommend
```suggestion
    assert.strictEqual(resResources.some(resource => utils.types.has(resource, `errorStatusCode`)), false)
```